drivers/shared/executor/cleanup.go

```diff
@@ -41,13 +41,23 @@ func (b *cleanupHandle) serialize() []byte {
 	return bytes
 }
 
-func processStartTime(pid int) uint64 {
-	stat, err := system.Stat(pid)
+// processStartTime attempts to determine the create time of a given process.
+func processStartTime(pid int32) (uint64, error) {
+	ps, err := process.NewProcess(pid)
 	if err != nil {
-		return 0
+		return 0, fmt.Errorf("Failed to find process: %v", err)
 	}
 
-	return stat.StartTime
+	st, err := ps.CreateTime()
+	if err != nil {
+		return 0, fmt.Errorf("Failed to find process start time: %v", err)
+	}
+
+	if st < 0 {
+		return 0, fmt.Errorf("Process started before unix epoch, or overflowed int64")
```

Comment:
I think overflowing is actually ok - given we only want equality tests and we can assume that the same value would overflow in the same way.  Granted that shouldn't happen in like 292 million years, it's not a big deal ;-).